### ISSUE TITLE
Database Media Storage - Design Config fails to save transactional email logo correctly

### ISSUE THREAD

**Issue by gwharton** [magento/magento2#21672]
### Preconditions (*)
1. 2.2-develop or 2.3-develop

### Steps to reproduce (*)
1. Setup Magento
2. Change media storage mode to database
3. Synchronize
4. Save Settings
5. Content->Configuration->Edit->Transactional Emails
6. Upload New Image (but don't save)
7. Verify temporary email logo exists in pub/media
8. Verify temporary email logo exists in database
9. Save Config
10. Verify email logo exists in pub/media
11. Verify email logo exists in database

### Expected result (*)
1. After saving the config, the email logo image should be present in both database and filesystem

### Actual result (*)
1. Email logo is not present in database

Step 7
```
www-data@dev:~/dev2$ find pub/media -name *.jpg
pub/media/tmp/design/file/Forklift.200x100.jpg
```
Step 8
```
www-data@dev:~/dev2$ mysql dev2_magento -e "select filename,directory from media_storage_file_storage"
+----------------------------------+-----------------+
| filename                         | directory       |
+----------------------------------+-----------------+
| preview_image_5c85077ebb298.jpeg | theme/preview   |
| preview_image_5c85077ed85e1.jpeg | theme/preview   |
| Forklift.200x100.jpg             | tmp/design/file |
+----------------------------------+-----------------+
```
Step 10
```
www-data@dev:~/dev2$ find pub/media -name *.jpg
pub/media/email/logo/stores/1/Forklift.200x100.jpg
```
Step 11
```
www-data@dev:~/dev2$ mysql dev2_magento -e "select filename,directory from media_storage_file_storage"
+----------------------------------+-----------------+
| filename                         | directory       |
+----------------------------------+-----------------+
| preview_image_5c85077ebb298.jpeg | theme/preview   |
| preview_image_5c85077ed85e1.jpeg | theme/preview   |
| Forklift.200x100.jpg             | tmp/design/file |
+----------------------------------+-----------------+
```
Step 11 has failed. The file is still in the database in the temporary location. It was not changed when the config was saved.

**Comment by magento-engcom-team** [comment]
Hi @gwharton. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me 2.3-develop instance` - upcoming 2.3.x release

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@gwharton do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

-  _Next steps are available in case you are a member of **Community Maintainers**._

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-backlog-nazar 
 Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-98682`, `MAGETWO-98683` were created

**Issue Available**: @engcom-backlog-nazar, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by gwharton** [comment]
#21675 and #21676 have been merged. Closing.

**Comment by magento-engcom-team** [comment]
Hi @gwharton. Thank you for your report. 
The issue has been fixed in magento/magento2#21674 by @gwharton in 2.3-develop branch
Related commit(s):
 - [c764ca310cdfa890caf6fc144da2a3aa5861cbee](https://github.com/magento/magento2/commit/c764ca310cdfa890caf6fc144da2a3aa5861cbee)
 - [057c13d66dc67c8291662ec5be77b87d26db781f](https://github.com/magento/magento2/commit/057c13d66dc67c8291662ec5be77b87d26db781f)
 - [1c05b797c32d2df6b32f1baf332132e8b7e47e23](https://github.com/magento/magento2/commit/1c05b797c32d2df6b32f1baf332132e8b7e47e23)

The fix will be available with the upcoming 2.3.3 release.

**Comment by magento-engcom-team** [comment]
Hi @gwharton. Thank you for your report. 
The issue has been fixed in magento/magento2#21673 by @gwharton in 2.2-develop branch
Related commit(s):
 - [abf66b8f9f6fec9fe9276fc07f6a477b3e486b22](https://github.com/magento/magento2/commit/abf66b8f9f6fec9fe9276fc07f6a477b3e486b22)
 - [36395ef4dc41c455682c9dfe697ec142e1f62408](https://github.com/magento/magento2/commit/36395ef4dc41c455682c9dfe697ec142e1f62408)
 - [21a8de11f7c92f656819d495888e4168419c3dc1](https://github.com/magento/magento2/commit/21a8de11f7c92f656819d495888e4168419c3dc1)

The fix will be available with the upcoming 2.2.10 release.
